core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -365,55 +365,65 @@ class WorkspaceService(protected val userInfo: UserInfo,
-                    else None
-                  }
+  private def gatherWorkflowsToAbortAndSetStatusToAborted(workspaceName: WorkspaceName, workspaceContext: Workspace) = {
+    dataSource.inTransaction { dataAccess =>
+      for {
+        // Gather any active workflows with external ids
+        workflowsToAbort <- dataAccess.workflowQuery.findActiveWorkflowsWithExternalIds(workspaceContext)
+
+        //If a workflow is not done, automatically change its status to Aborted
+        _ <- dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { recs =>
+          recs.collect {
+            case wf if !WorkflowStatuses.withName(wf.status).isDone =>
+              dataAccess.workflowQuery.updateStatus(wf, WorkflowStatuses.Aborted) { status =>
+                if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceName, wf.submissionId))(status))
+                else None
               }
-            }
+          }
+        }
+      } yield {
+        workflowsToAbort
+      }
+    }
+  }
 
-            // Delete components of the workspace
-            _ <- dataAccess.submissionQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
-            _ <- dataAccess.methodConfigurationQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
-            _ <- dataAccess.entityQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+  private def deleteWorkspaceTransaction(workspaceName: WorkspaceName, workspaceContext: Workspace) = {
+    dataSource.inTransaction { dataAccess =>
+      for {
+        // Delete components of the workspace
+        _ <- dataAccess.submissionQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.methodConfigurationQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.entityQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
 
-            // Delete the workspace
-            _ <- dataAccess.workspaceQuery.delete(workspaceName)
+        // Schedule bucket for deletion
+        _ <- dataAccess.pendingBucketDeletionQuery.save(PendingBucketDeletionRecord(workspaceContext.bucketName))
 
-            // Schedule bucket for deletion
-            _ <- dataAccess.pendingBucketDeletionQuery.save(PendingBucketDeletionRecord(workspaceContext.bucketName))
+        // Delete the workspace
+        _ <- dataAccess.workspaceQuery.delete(workspaceName)
+      } yield ()
+    }
+  }
 
-          } yield {
-            workflowsToAbort
-          }
+  private def deleteWorkspace(workspaceName: WorkspaceName, workspaceContext: Workspace): Future[PerRequestMessage] = {
+    for {
+      _ <- requesterPaysSetupService.revokeAllUsersFromWorkspace(workspaceContext) recoverWith {
+        case t:Throwable => {
+          logger.warn(s"Unexpected failure deleting workspace (while revoking 'requester pays' users) for workspace `${workspaceName}`", t)
```

Comment:
Can you also log the workspace name and/or context?

Reply:
Also the "unexpected..." stuff is rather verbose. Doesn't logging give us where the thing was logged? How about
`"Failed to revoke all users from workspace"`, then include which workspace?

Reply:
`WorkspaceName` contains billing-project/workspace, which I think is all we need from here.

the reason for the verbose logging is for ease of search. We have log sinks to catch all of these errors starting with `Unexpected failure deleting` (CA-1470) to help with debugging for CA-1206.

Reply:
Thanks for the explanation.